File: solutions/812-rotate-string/rotate-string.java
class Solution {
    public boolean rotateString(String s, String goal) {
        // Get lengths of both strings
        int n = s.length();
        int m = goal.length();
        
        // If lengths are different, rotation is impossible
        if (n != m) {
            return false;
        }
        
        // Convert to char array for manipulation
        char[] sArray = s.toCharArray();
        
        // Try all possible rotations
        for (int i = 0; i < n; i++) {
            // If current string matches goal, we found a valid rotation
            if (String.valueOf(sArray).equals(goal)) {
                return true;
            }
            
            // Perform one left rotation
            char a = sArray[0];
            for (int j = 0; j < n-1; j++) {
                sArray[j] = sArray[j+1];
            }
            sArray[n-1] = a;
        }
        
        return false;
    }
}

File: solutions/812-rotate-string/rotate-string.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        # Get lengths of both strings
        n = len(s)
        m = len(goal)
        
        # If lengths are different, rotation is impossible
        if n != m:
            return False
            
        # Convert string to list for easier manipulation
        s_list = list(s)
        
        # Try all possible rotations
        for i in range(n):
            # If current string matches goal, we found a valid rotation
            if ''.join(s_list) == goal:
                return True
                
            # Perform one left rotation
            a = s_list[0]
            for j in range(n-1):
                s_list[j] = s_list[j+1]
            s_list[n-1] = a
            
        return False